service/controller/internal/cloudconfig/tccpn.go

```diff
@@ -4,36 +4,43 @@ import (
 	"context"
 	"encoding/base64"
 	"fmt"
+	"net"
 
 	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2"
-	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
 	"github.com/giantswarm/certs"
 	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_5_2_0"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/randomkeys"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
 	cloudconfig "github.com/giantswarm/aws-operator/service/controller/internal/cloudconfig/template"
-	"github.com/giantswarm/aws-operator/service/controller/internal/encrypter/vault"
 	"github.com/giantswarm/aws-operator/service/controller/key"
 )
 
 type TCCPNConfig struct {
-	Config Config
+	Config    Config
+	G8sClient versioned.Interface
```

Comment:
The idea here initially was to have a common config. Is it that this diverges now? I see we list resources in the render function. This is not cool since rendering should only be about templates and not making network calls. So whatever we need to render templates should be injected via configuration. I assume the client then also goes away and the config is "clean" again.

Reply:
for HA masters we anyway need to diverge more since for masters we need 3 unique ignitions and each ignition needs custom masterID (we need to somehow distinguish between each master) so I thnik this is unavoidable. 

Either we need to change the Render function  or something else.